### PR TITLE
Show details when click on marker

### PR DESCRIPTION
**Pull request by korinsek**
On marker click starts new activity.

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -24,6 +26,7 @@
             android:name="com.google.android.geo.API_KEY"
             android:value="@string/api_key" />
 
+        <activity android:name=".ui.details.DetailsActivity"></activity>
```
Comment (tmislej) [inline app/src/main/AndroidManifest.xml]: You can use the xml closing tag here `/>` and lock activity into portrait mode.

File: app/src/main/java/com/tine/coffeeshops/ui/details/DetailsMvp.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+/**
+ * Created by denis on 06/09/17.
+ */
+
+public interface DetailsMvp {
+
+    interface View {
+
+        void onReady();
```
Comment (tmislej) [inline app/src/main/java/com/tine/coffeeshops/ui/details/DetailsMvp.java]: for what will this be used for?

File: app/src/main/java/com/tine/coffeeshops/ui/details/DetailsPresenter.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import android.content.Context;
+
+/**
+ * Created by denis on 06/09/17.
+ */
+
+public class DetailsPresenter implements DetailsMvp.Presenter {
+
+    private final DetailsMvp.View view;
+    private final Context context;
```
Comment (tmislej) [inline app/src/main/java/com/tine/coffeeshops/ui/details/DetailsPresenter.java]: Why do you need context?

File: app/src/main/java/com/tine/coffeeshops/ui/main/MainActivity.java [hunk 4]
```diff
@@ -127,4 +129,10 @@ protected void onCreate(Bundle savedInstanceState) {
     public void onRequestPermissionsResult(int requestCode, String permissions[], int[] grantResults) {
         presenter.onRequestPermissionsResult(requestCode, permissions, grantResults);
     }
+
+    @Override
+    public void openDetailsView() {
+        Intent i = new Intent(this, DetailsActivity.class);
```
Comment (tmislej) [inline app/src/main/java/com/tine/coffeeshops/ui/main/MainActivity.java]: This is ok, but we use a bit different pattern for generating intents. Your activity should implement a public static method which will return the intent for that activity. Any data that has to be passed to activity is passed via method params. We usually name that method as `getIntent` or `newIntent`. Use whichever you prefer.

File: app/src/main/java/com/tine/coffeeshops/ui/main/map/CoffeeShopsMapModule.java [hunk 5]
```diff
@@ -25,10 +26,10 @@ public CoffeeShopsMapModule(CoffeeShopsMapMvp.View view) {
     }
 
     @ViewScope @Provides CoffeeShopsMapMvp.Presenter providePresenter(CoffeeShopsMapMvp.View view,
-            PlacesApiService placesApiService, Resources resources, MainThreadManager mainThreadManager,
-            LocationManager locationManager) {
+                                                                      PlacesApiService placesApiService, Resources resources, MainThreadManager mainThreadManager,
```
Comment (tmislej) [inline app/src/main/java/com/tine/coffeeshops/ui/main/map/CoffeeShopsMapModule.java]: remove this white space.

File: app/src/main/java/com/tine/coffeeshops/ui/main/map/CoffeeShopsMapPresenter.java [hunk 6]
```diff
@@ -26,19 +33,21 @@
...
+    private final MainMvp.Presenter parentPresenter;
 
     private ClusterManager<UiPlace> clusterManager;
 
     private boolean isMapReady = false;
     private Subscription locationSubscription;
 
     public CoffeeShopsMapPresenter(CoffeeShopsMapMvp.View view, PlacesApiService placesApiService,
-            Resources resources, MainThreadManager mainThreadManager, LocationManager locationManager) {
+                                   Resources resources, MainThreadManager mainThreadManager, LocationManager locationManager, MainMvp.Presenter parentPresenter) {
```
Comment (tmislej) [inline app/src/main/java/com/tine/coffeeshops/ui/main/map/CoffeeShopsMapPresenter.java]: same here

File: app/src/main/java/com/tine/coffeeshops/ui/main/map/CoffeeShopsMapPresenter.java [hunk 7]
```diff
@@ -26,19 +33,21 @@
...
         this.view = view;
         this.placesApiService = placesApiService;
         this.resources = resources;
         this.mainThreadManager = mainThreadManager;
         this.locationManager = locationManager;
+        this.parentPresenter = parentPresenter;
```
Comment (tmislej) [inline app/src/main/java/com/tine/coffeeshops/ui/main/map/CoffeeShopsMapPresenter.java]: 👌

File: app/src/main/java/com/tine/coffeeshops/ui/main/map/CoffeeShopsMapModule.java (hunk 5 above)
Comment (korinsek) [reply app/src/main/java/com/tine/coffeeshops/ui/main/map/CoffeeShopsMapModule.java]: Code style - I'm waiting for code style template.

File: app/src/main/java/com/tine/coffeeshops/ui/details/DetailsActivity.java [hunk 8]
```diff
@@ -32,4 +29,8 @@ public void onReady() {
 
         return component;
     }
+
+    public static Intent getIntent(Context context) {
+        return new Intent(context, DetailsActivity.class);
```
Comment (tmislej) [inline app/src/main/java/com/tine/coffeeshops/ui/details/DetailsActivity.java]: 👌

File: app/src/main/java/com/tine/coffeeshops/ui/main/map/CoffeeShopsMapModule.java (hunk 5 above)
Comment (tmislej) [reply app/src/main/java/com/tine/coffeeshops/ui/main/map/CoffeeShopsMapModule.java]: 👍

**Pull request merged**